collections/_article/godot-xr-update-oct-2024.md

```diff
@@ -0,0 +1,136 @@
+---
+title: "Godot XR update - October 2024"
+excerpt: "Looking back at a year of XR development."
+categories: ["progress-report"]
+author: Bastiaan Olij
+image: /storage/blog/covers/october-2024-xr-progress-update.webp
+image_caption_description: "Untitled penguin VR project by Zi Ye"
+date: 2024-10-01 11:00:00
+---
+
+It has been a very busy year on the XR front and seeing we haven’t given a proper update for well over a year, it’s time to look back at what has happened since our last update post.
+We touched on some of these in the Godot 4.3 release blog but we’re doing a bit of a deep dive here.
+
+We’re planning on more XR focused posts in the near future, we recently [announced the XR editor](https://godotengine.org/article/godot-editor-horizon-store-early-access-release/) but we have more news on the way.
+
+## Improvements in Godot's XR architecture
+
+A lot of work has gone into improving the XR implementation in Godot.
+
+### Improved frame timing logic
+
+When OpenXR was originally implemented in Godot 3 there were limitations to the order in which certain actions took place. Most of the communication with OpenXR was tied into the rendering engine resulting in Godot relying more on predictive algorithms to position tracked entities such as controllers. Originally Godot 4 inherited this approach.
+
+As part of the Godot 4.3 release much of this logic was redone to interface with OpenXR correctly. Godot now correctly informs OpenXR when it starts processing a new frame and obtains pose data for tracked entities with predicted positioning for the current frame.
+
+### Preparing for support rendering on a separate thread
+
+As part of implementing the improved frame timing logic, we did a large cleanup of the XR code to make sure it runs properly when rendering happens on a separate thread. Especially for XR applications this brings us closer to ensuring a steady framerate even on stand alone devices.
+
+There is more work that needs to be done in Godot 4.4 to make this fully viable but the foundations are in place.
+
+### Standardisation for AR and passthrough
+
+Passthrough is a mechanism where video cameras on a VR headset record the real world and present this real world view as the background inside of our HMD turning the headset into an AR device. Different headsets implement this feature in different ways and this caused difficulties for developers who were targeting multiple devices.
+
+While passthrough allows some neat things not possible with optical AR devices, for most AR focussed applications developers should not be tasked with writing different code to support all these devices.
+
+We have thus standardised this approach through utilising OpenXRs environment blend modes and moving platforms specific code into the OpenXR vendors plugin.
```

Comment:
In the code and docs we try to use US spelling rather than UK, but I don't know if that matters in a blog post? Probably not. But if it does, this would need to be:

```suggestion
We have thus standardized this approach through utilising OpenXRs environment blend modes and moving platforms specific code into the OpenXR vendors plugin.
```